### PR TITLE
[WIP] Quick start guide for CAPBM

### PR DESCRIPTION
**Pull request by Xenwar**
This PR adds content that shows the architecture and components (controller and CRs) involved with CAPBM. 
It also show the flow of information from CR to CR with the help of different controllers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,120 @@ API used with this `cluster-api` provider. You can also see the [cluster
 deployment workflow](docs/deployment_workflow.md) for the outline of the
 deployment process.
 
+## Components
+The ```cluster-api-provider-baremetal (CAPBM)``` operator is responsible watching and reconciling multiple resources. However, it is important to see other controllers and custom resources (CRs) involved in the process. The ultimate goal of the interaction between the controllers and CRs is to provision a kubernetes cluster on Bare Metal machines. To that end, the controllers take different roles.
+
+The following diagram shows the different controllers and CRs involved. The CAPI, CABPK and CABPK controllers are beyond the scope of this document. With respect to CAPBM, we focus on what CRS it `watches` and `reconciles`. In the following diagram arrows in <span style="color:black">**black**</span> show which CRs a controller `reconciles` while the one in <span style="color:red">**red**</span> show a `watching` relationship.
```
Comment (Jaakko-Os) [inline README.md]: duplicate CABPK
should be CAPBM?

File: README.md [hunk 2]
```diff
@@ -24,6 +24,120 @@ API used with this `cluster-api` provider. You can also see the [cluster
...
+
+![components](docs/components.png)
+
+As shown in the above Components' diagram, CAPBM is watching for changes in `Machine` CR and upon change, it makes changes on the `BareMetal Machine` CR. Similarly, it watches a `Cluster` CR and makes changes on a related `BareMetal Cluster` CR.
+
+The left most components, BMO controller and BareMetalHost(BMH) Cr, are the closest to the Bare Metal physical machine. If one wants to changes the state of the Bare Metal machine, they modify the BMH CR. Upon change to the BMH, BMO interacts with Ironic to make changes on the Bare Metal machine.
+
+The above logis in the opposite direction as well. If there is any change on the Bare Meta machine, BMO learns about the change(s) via Ironic and a chain effect starts. BMO, after learning about the changes, makes the required changes on the BMH CR and CAPM, upon seeing the changes on a BMH resource, it makes the relevant changes on the BareMetalMachine CR.
```
Comment (Jaakko-Os) [inline README.md]: CAPM?

**Review by Jaakko-Os — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -24,6 +24,120 @@ API used with this `cluster-api` provider. You can also see the [cluster
...
+
+As discussed above, the management of Bare Metal Machines requires the interaction of multiple controllers via multiple CRs. However, the interaction is performed on a specified number of fields on each CR. i.e. A controller `watches` a specified number of fields in each CR and makes changes on a set of fields. The following diagram specifies the mapping of fields between the CRs shown in the components' diagram above.
```
Comment (fmuyassarov) [inline README.md]: Bare Metal Machines ? Bare Metal machines ? it is not the same everywhere

File: README.md [hunk 4]
```diff
@@ -24,6 +24,120 @@ API used with this `cluster-api` provider. You can also see the [cluster
...
+
+The following diagram shows the different controllers and CRs involved. The CAPI, CABPK and CABPK controllers are beyond the scope of this document. With respect to CAPBM, we focus on what CRS it `watches` and `reconciles`. In the following diagram arrows in <span style="color:black">**black**</span> show which CRs a controller `reconciles` while the one in <span style="color:red">**red**</span> show a `watching` relationship.
+
+![components](docs/components.png)
+
+As shown in the above Components' diagram, CAPBM is watching for changes in `Machine` CR and upon change, it makes changes on the `BareMetal Machine` CR. Similarly, it watches a `Cluster` CR and makes changes on a related `BareMetal Cluster` CR.
+
+The left most components, BMO controller and BareMetalHost(BMH) Cr, are the closest to the Bare Metal physical machine. If one wants to changes the state of the Bare Metal machine, they modify the BMH CR. Upon change to the BMH, BMO interacts with Ironic to make changes on the Bare Metal machine.
```
Comment (fmuyassarov) [inline README.md]: controller and BareMetalHost(BMH) **CR**

File: README.md (hunk 1 above)
Comment (fmuyassarov) [inline README.md]: on what **CRs** it

File: README.md [hunk 5]
```diff
@@ -24,6 +24,120 @@ API used with this `cluster-api` provider. You can also see the [cluster
 deployment workflow](docs/deployment_workflow.md) for the outline of the
 deployment process.
 
+## Components
+The ```cluster-api-provider-baremetal (CAPBM)``` operator is responsible watching and reconciling multiple resources. However, it is important to see other controllers and custom resources (CRs) involved in the process. The ultimate goal of the interaction between the controllers and CRs is to provision a kubernetes cluster on Bare Metal machines. To that end, the controllers take different roles.
```
Comment (fmuyassarov) [inline README.md]: responsible **for** watching

**Review by fmuyassarov — CHANGES_REQUESTED** [review]

**Pull request merged**